src/esm_parser/esm_parser.py

```diff
@@ -1038,7 +1065,10 @@ def add_entry_to_chapter(
                     add_chapter.split(".")[-1].replace("add_", "")
                 ]
                 # Add the entries
-                mod_list += add_entries
+                if isinstance(add_entries, list):
+                    mod_list.extend(list(flatten_nested_lists(add_entries)))
+                else:
+                    mod_list.append(add_entries)
```

Comment:
Not sure this is needed...? Isn't `mod_list` already defined behind a `isinstance` check against lists? From my point of view, this only needs to be:
```suggestion
                    mod_list.extend(list(flatten_nested_lists(add_entries)))
```

Reply:
I think `mod_list` is the list into which you add, which is already under a `isinstance`, but `add_entries` is what you want to add to that first list, and that could be a string or a list to be flattened, so I think the whole if and else is needed here

Reply:
Yes, I controlled that and you are right. I removed the redundant checks and only kept the extend. Moreover, I added better error handling by `user_error`. If an uncompatible type is tried to merge now we will have:
```
ERROR: Type mismatch
--------------------
Can not add a variable of incompatible type int to mylist3
```
instead of a long traceback with `TypeError: Something is wrong` which is not very explanatory.